thumbor/handlers/__init__.py

```diff
@@ -281,7 +281,7 @@ def finish_request(self, context, result_from_storage=None):
             context.config.RESULT_STORAGE_STORES_UNSAFE or not context.request.unsafe)
 
         def inner(future):
-            results, content_type = future.result()
+            results, content_type = (future.result()[0], BaseEngine.get_mimetype(future.result()[0]))
```

Comment:
@guilhermef I'll keep digging, but I tried to put the BaseEngine call into the `if context.request.format` block right here https://github.com/okor/thumbor/blob/master/thumbor/handlers/__init__.py#L246 but it's not being executed. I added a debug print statement and `context.request.format` is returning None. But the plugin won't even run if that was the case (it's definitely running and modifying the MIME type). So something seems amiss. Any ideas?